streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -190,10 +145,52 @@ public String toString(final String indent) {
         return sb.toString();
     }
 
+    protected Map<TopicPartition, Long> recordCollectorOffsets() {
+        return Collections.emptyMap();
+    }
+
+    protected void initializeOffsetLimits() {
```

Comment:
Yes. Just moved some methods in the third commit to "cluster" them a little bit in the code -- right now it's all scattered around the whole file and I have always are hard time to keep an overview (maybe it's just me...)